### PR TITLE
Added keyboard shortcuts management

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: apps/BraynsViewer/BraynsViewer.h [hunk 1]
```diff
@@ -39,22 +39,15 @@ class BraynsViewer : public BaseWindow
...
-     * '*' : Displays parameters helper in the console
-     *
-     * @param key key pressed
-     * @param where location of the window cursor
-     */
-    void keypress(char key, const Vector2f& where) final;
-
 private:
+
+    virtual void _registerKeyboardShortcuts() final;
```
Comment (tribal-tec) [inline apps/BraynsViewer/BraynsViewer.h]: either virtual or final

File: brayns/common/scene/Scene.cpp [hunk 2]
```diff
@@ -542,7 +542,7 @@ VolumeHandlerPtr Scene::getVolumeHandler()
                         {
                             boost::filesystem::path fileExtension =
                                 dirIter->path( ).extension( );
-                            if( fileExtension==".raw" )
+                            //if( fileExtension==".raw" )
```
Comment (tribal-tec) [inline brayns/common/scene/Scene.cpp]: on purpose?

File: brayns/io/MorphologyLoader.cpp [hunk 3]
```diff
@@ -134,8 +134,8 @@ bool MorphologyLoader::_importMorphology(
             const float radius =
                 ( _geometryParameters.getRadiusCorrection() != 0.f ?
                 _geometryParameters.getRadiusCorrection() :
-                soma.getMeanRadius() *
-                    _geometryParameters.getRadiusMultiplier() );
+                soma.getMeanRadius() /* *
+                    _geometryParameters.getRadiusMultiplier()*/ );
```
Comment (tribal-tec) [inline brayns/io/MorphologyLoader.cpp]: dito?

File: brayns/io/NESTLoader.cpp [hunk 4]
```diff
@@ -228,6 +229,10 @@ bool NESTLoader::_load( const float timestamp )
     BRAYNS_DEBUG << "Loading spikes at timestamp: " << timestamp
                  << " (" << start << " ms)" << std::endl;
 
+
+    char filename[1024];
+    sprintf( filename, "/home/favreau/medias/volumes/spikes/10pct/events/spikes_%05d.events", int(timestamp*10) );
```
Comment (tribal-tec) [inline brayns/io/NESTLoader.cpp]: dito?

File: plugins/extensions/plugins/DeflectPlugin.cpp [hunk 5]
```diff
@@ -20,188 +20,197 @@
...
+    ApplicationParameters& applicationParameters =
+        _brayns.getParametersManager().getApplicationParameters();
+    const bool deflectEnabled = applicationParameters.getDeflectEnabled();
+
+    if( _stream && _stream->isConnected() && !deflectEnabled )
+    {
+        BRAYNS_INFO << "Closing Deflect stream" << std::endl;
+        delete _stream.get();
+        _stream.release();
+        _stream = nullptr;
```
Comment (tribal-tec) [inline plugins/extensions/plugins/DeflectPlugin.cpp]: ouch! _stream.reset() should be enough

File: plugins/extensions/plugins/DeflectPlugin.cpp [hunk 6]
```diff
@@ -20,188 +20,197 @@
...
-    deflectImage.compressionQuality = _compressionQuality;
     if( swapXAxis )
-        deflect::ImageWrapper::swapYAxis(
-            (void*)imageData, windowSize.x(), windowSize.y(), 4);
+        deflect::ImageWrapper::swapYAxis( (void*)imageData, windowSize.x(), windowSize.y(), 4);
 
-    bool success = _stream->send(deflectImage);
+    // TODO: Big Memory Leak on every send. FIX NEEDED!!!!!
+    const bool success = _stream->send( deflectImage );
     _stream->finishFrame();
```
Comment (tribal-tec) [inline plugins/extensions/plugins/DeflectPlugin.cpp]: also evaluate result of finishFrame(). We typically do result = send() && finish();

File: apps/BraynsViewer/BraynsViewer.h (hunk 1 above)
Comment (favreau) [inline apps/BraynsViewer/BraynsViewer.h]: final :-) Thanks

File: brayns/io/NESTLoader.cpp (hunk 4 above)
Comment (favreau) [inline brayns/io/NESTLoader.cpp]: Oops, not on purpose. Thanks again.

File: plugins/extensions/plugins/DeflectPlugin.cpp (hunk 5 above)
Comment (favreau) [inline plugins/extensions/plugins/DeflectPlugin.cpp]: I should have read 'smart pointers for dummies'. Fixed.

File: plugins/extensions/plugins/DeflectPlugin.cpp (hunk 6 above)
Comment (favreau) [inline plugins/extensions/plugins/DeflectPlugin.cpp]: Done.

**Comment by favreau** [comment]
Retest this please

**Comment by favreau** [comment]
Retest this please

**Comment by favreau** [comment]
Retest this please

**Pull request merged**